From: Radim Krcmar <rkrcmar@redhat.com>
Date: Fri, 3 Sep 2010 12:24:13 -0400
Subject: [pci] xen: disable broken msi/msix on ia64 xen
Message-id: <1283516653-23235-1-git-send-email-rkrcmar@redhat.com>
Patchwork-id: 28062
O-Subject: Re: [RHEL5.6 PATCH] pci: forbid msi on ia64 xen, fix caused breakage
Bugzilla: 518463
RH-Acked-by: Jarod Wilson <jarod@redhat.com>
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Andrew Jones <drjones@redhat.com>

bug: https://bugzilla.redhat.com/show_bug.cgi?id=518463

MSI and MSI-X are not working on ia64 with xen, polluting dmesg with failures.

Upstream has it fixed, but the changeset 18686 would require whole vt-d, which
was deemed unwise and no customer requested this functionality.

This patch disables MSI and MSI-X unilaterally on ia64 xen, eliminating the
dmesg failure spew.
brew: https://brewweb.devel.redhat.com/taskinfo?taskID=2732463


diff --git a/drivers/pci/msi-xen.c b/drivers/pci/msi-xen.c
index afa83d4..1302052 100644
--- a/drivers/pci/msi-xen.c
+++ b/drivers/pci/msi-xen.c
@@ -574,6 +574,10 @@ int pci_enable_msi(struct pci_dev* dev)
 	int pos, temp, status = -EINVAL;
 	struct msi_dev_list *msi_dev_entry = get_msi_dev_pirq_list(dev);
 
+#ifdef CONFIG_IA64
+	return -EINVAL;
+#endif
+
 	if (!pci_msi_enable || !dev)
 		return status;
 
@@ -686,6 +690,10 @@ int pci_enable_msix(struct pci_dev* dev, struct msix_entry *entries, int nvec)
 	u16 control;
 	struct msi_dev_list *msi_dev_entry = get_msi_dev_pirq_list(dev);
 
+#ifdef CONFIG_IA64
+	return -EINVAL;
+#endif
+
 	if (!pci_msi_enable || !dev || !entries)
 		return -EINVAL;
 
